code/src/reducers/projects.js

```diff
@@ -0,0 +1,75 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export const projects = createSlice({
+  name: "projects",
+  initialState: {
+    project: [],
+  },
+
+  reducers: {
```

Comment:
So impressed to see all actions the user can do with the app. This results in a long list of reducer actions. Wouldn't it be better to use the immutability approach that can "bring increased performance to your app" (https://redux.js.org/faq/immutable-data#what-are-the-benefits-of-immutability). For example I see that you chose to use the 'push' and 'slice' method (mutability approach). As covered in one of the lecture we could write an 'immutable' code instead. For example you could replace line 19 with `state.project = [...state.project, action.payload];`. This was just an idea. The app still works good as it is. As a user I don't seem to find any bugs or so to it.

Reply:
Yeah, I remember when Maksy was talking about the difference between mutability and immutability, and that we're only able to use the mutability thanks to immer. At that point (the Wednesday curse I call it), I didn't really have the energy to re-write it as immutable. Originally I had four stores - I was storing my tasks separately from my project lists, and including a project ID with each task. Daniel suggested in town hall that I nest my tasks within the individual projects, so I was already quite pressured at that time, because I had to re-write all my reducers, as well as selectors and dispatchers to achieve just a single reducer that contained both projects and tasks.